pkg/replicaset/replicaset.go

```diff
@@ -0,0 +1,446 @@
+
+		return builder
+	}
+
+	if builder.Definition.Labels == nil {
+		builder.Definition.Labels = map[string]string{}
+	}
+
+	builder.Definition.Labels[labelKey] = labelValue
+
+	return builder
+}
+
+// WithNodeSelector applies nodeSelector to the replicaset definition.
+func (builder *Builder) WithNodeSelector(nodeSelectorKey, nodeSelectorValue string) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying nodeSelector %s: %s to replicaset %s in namespace %s",
+		nodeSelectorKey, nodeSelectorValue, builder.Definition.Name, builder.Definition.Namespace)
+
+	if nodeSelectorKey == "" {
+		glog.V(100).Infof("The 'nodeSelectorKey' of the replicaset is empty")
+
+		builder.errorMsg = "can not apply empty nodeSelectorKey"
+
+		return builder
+	}
+
+	if builder.Definition.Spec.Template.Spec.NodeSelector == nil {
+		builder.Definition.Spec.Template.Spec.NodeSelector = map[string]string{}
+	}
+
+	builder.Definition.Spec.Template.Spec.NodeSelector[nodeSelectorKey] = nodeSelectorValue
+
+	return builder
+}
+
+// WithVolume defines Volume of replicaset under ContainerTemplateSpec.
+func (builder *Builder) WithVolume(dsVolume corev1.Volume) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if dsVolume.Name == "" {
+		glog.V(100).Infof("The Volume name parameter is empty")
+
+		builder.errorMsg = "Volume name parameter is empty"
+
+		return builder
+	}
+
+	glog.V(100).Infof("Adding volume %s for replicaset %s container template in namespace %s",
+		dsVolume.Name, builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Definition.Spec.Template.Spec.Volumes = append(
+		builder.Definition.Spec.Template.Spec.Volumes,
+		dsVolume)
+
+	return builder
+}
+
+// WithAdditionalContainerSpecs appends a list of container specs to the replicaset definition.
+func (builder *Builder) WithAdditionalContainerSpecs(specs []corev1.Container) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Appending a list of container specs %v to replicaset %s in namespace %s",
+		specs, builder.Definition.Name, builder.Definition.Namespace)
+
+	if len(specs) == 0 {
+		glog.V(100).Infof("The container specs are empty")
+
+		builder.errorMsg = "cannot accept empty list as container specs"
+
+		return builder
+	}
+
+	if builder.Definition.Spec.Template.Spec.Containers == nil {
+		builder.Definition.Spec.Template.Spec.Containers = specs
+	} else {
+		builder.Definition.Spec.Template.Spec.Containers = append(builder.Definition.Spec.Template.Spec.Containers, specs...)
+	}
+
+	return builder
+}
+
+// Exists checks whether the given replicaset exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if replicaset %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
```

Comment:
Please use standtart k8s client instead of generic:
```
	builder.Object, err = builder.apiClient.ReplicaSets(builder.Definition.Namespace).Get(
		context.TODO(), builder.Definition.Name, metav1.GetOptions{})

	return err == nil || !k8serrors.IsNotFound(err)
```